1. Пусть $X∼F(x)$ — случайная величина с произвольным распределением, неизвестным математическим ожиданием EX и известной дисперсией DX. Как выглядит доверительный интервал для EX с приближённым уровнем доверия 100(1−α)%?

A.         $\bar{X}_{n}\pm z_{1-\frac{\alpha}{2}}\frac{DX}{\sqrt{n}}$ </br>
</br>
B.         $\bar{X}_{n}\pm t_{n-1, 1-\frac{\alpha}{2}}\sqrt{\frac{DX}{n}}$ </br>
</br>
C.         $\bar{X}_{n}\pm z_{1-\frac{\alpha}{2}}{\sqrt\frac{DX}{n}}$ </br>
</br>
D.         $\bar{X}_{n}\pm z_{1-\alpha}\sqrt{\frac{DX}{n}}$ </br>

In [ ]:
# 1. C

# interval = z_value * (st_dev / np.sqrt(sample_size))
# conf_int = (sample_mean - interval, sample_mean + interval)

2. Для 61 большого города в Англии и Уэльсе известны средняя годовая смертность на 100000 населения (по данным 1958–1964) и концентрация кальция в питьевой воде (в частях на миллион). Чем выше концентрация кальция, тем жёстче вода. Города дополнительно поделены на северные и южные.

In [ ]:
import numpy as np
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import random
from tqdm import tqdm
plt.style.use('ggplot')
from PIL import Image

In [ ]:
# Load water.txt file
df = pd.read_csv('/content/water.csv', sep='\t')

In [ ]:
df.head()

location        town  mortality  hardness
0    South        Bath       1247       105
1    North  Birkenhead       1668        17
2    South  Birmingham       1466         5
3    North   Blackburn       1800        14
4    North   Blackpool       1609        18

Постройте 95% доверительный интервал для средней годовой смертности в больших городах. Чему равна его нижняя граница? Округлите ответ до 4 знаков после десятичной точки. </br>
 </br>
  </br>
   </br>
**Будьте осторожны при использовании метода std()!** Дело в том, что у объекта numpy он по умолчанию вычисляется как
$\sqrt{\frac{1}{n}\sum_{i=1}^n(X_i-\bar{X})}$, а у объекта pandas — как $\sqrt{\frac{1}{n-1}\sum_{i=1}^n(X_i-\bar{X})}$. </br>
 </br>
Интересует только второй вариант, несмещённая оценка стандартного отклонения. </br>
 </br>
Чтобы не думать всё время о том, правильно ли вычисляется в вашем случае std(), можно всегда использовать std(ddof=1) (ddof — difference in degrees of freedom), тогда нормировка всегда будет на n-1.

In [ ]:
# 2. Your answer here

def compute_ci(sample, st_dev):
    
    z_value = stats.norm.ppf(q=0.975)
    sample_size = len(sample)
    interval = z_value * (st_dev / np.sqrt(sample_size))
    conf_int = (sample.mean() - interval, sample.mean() + interval)
    
    return conf_int

In [ ]:
compute_ci(df['mortality'], df['mortality'].std(ddof=1))

(1477.0525038218827, 1571.2425781453305)

1477.0525

3. На данных из предыдущего вопроса постройте 95% доверительный интервал для средней годовой смертности по всем южным городам. Чему равна его верхняя граница? Округлите ответ до 4 знаков после десятичной точки.

In [ ]:
# 3. Your answer here
compute_ci(df[df.location == 'South']["mortality"], df[df.location == 'South']["mortality"].std(ddof=1))

(1322.8909479237761, 1430.7244366916086)

1430.7244

4. На тех же данных постройте 95% доверительный интервал для средней годовой смертности по всем северным городам. Пересекается ли этот интервал с предыдущим? Как вы думаете, какой из этого можно сделать вывод? 

In [ ]:
# 4. Your answer here
compute_ci(df[df.location == 'North']["mortality"], df[df.location == 'North']["mortality"].std(ddof=1))

(1588.2335708894593, 1678.9664291105405)

Mortality in South - [1322.8909, 1430.7244]

---


Mortality in North - [1588.2335, 1678.9664]

---
Mortality in North is much more than mortality in South


5. Пересекаются ли 95% доверительные интервалы для средней жёсткости воды в северных и южных городах?

A. Пересекаются </br>
B. Не пересекаются 

In [ ]:
# 5. Your answer here
print("North: ", compute_ci(df[df.location == 'North']["hardness"], df[df.location == 'North']["hardness"].std(ddof=1)))
print("South: ", compute_ci(df[df.location == 'South']["hardness"], df[df.location == 'South']["hardness"].std(ddof=1)))

North:  (21.741789085882083, 39.058210914117915)
South:  (54.255369325168935, 85.2830922132926)


6. Вспомним формулу доверительного интервала для среднего нормально распределённой случайной величины с дисперсией $\sigma^2$: </br>
$\bar{X}_{n}\pm z_{1-\frac{\alpha}{2}}\frac{\sigma}{\sqrt{n}}$ </br>
При $\sigma = 1$ какой нужен объём выборки, чтобы на уровне доверия 95% оценить среднее с точностью $\pm0.1$?

In [ ]:
# 6. Your answer here
z_value = stats.norm.ppf(q=0.975)
(z_value / 0.1)**2

384.14588206941244

385

7. Объём выборки, необходимый для построения доверительного интервала заданной ширины: </br>
A. увеличивается с уменьшением требуемой ширины, </br>
B. уменьшается с ростом дисперсии выборки, </br>
C. увеличивается с увеличением требуемой ширины, </br>
D. увеличивается с ростом дисперсии выборки, </br>
E. увеличивается с ростом $\alpha$, </br>
F. уменьшается с ростом $\alpha$. </br>

In [ ]:
# 7. Your answer here
# A, D, F

8. Давайте уточним правило трёх сигм. Утверждение: 99.7% вероятностной массы случайной величины $X∼N(\mu,\sigma^2)$ лежит в интервале $\mu±c⋅\sigma$. Чему равно точное значение константы c? Округлите ответ до четырёх знаков после десятичной точки.

In [ ]:
# 8. Your answer here
stats.norm.ppf(q=1-0.003/2)

2.9677379253417944

9. Пусть $X∼N(\mu,\sigma^2)$. Какое распределение имеет величина $\frac{\bar{X}_n - \mu}{S_n/\sqrt{n}}$? </br>
A. $St(n-1)$ </br>
</br>
B. $N(0, 1)$ </br>
</br>
C. $\chi_{n-1}^2$ </br>

In [ ]:
# 9. Your answer here
# A

10. Выберите все распределения с несимметричной функцией плотности: </br>

A. Фишера, </br>
B. Стьюдента, </br>
C. хи-квадрат, </br>
D. Гаусса

In [ ]:
# 10. Your answer here
# A, C